example/spec/helpers/discoverUtils.js

```diff
@@ -18,7 +18,7 @@ const uploadS3GranuleDataForDiscovery = async ({
   const granuleRegex = '^MOD09GQ\\.A[\\d]{7}\\.[\\w]{6}\\.006\\.[\\d]{13}$';
 
   const newGranuleId = randomStringFromRegex(granuleRegex);
-  await s3Data.map((file) => updateAndUploadTestFileToBucket({
+  await s3Data.map(async (file) => await updateAndUploadTestFileToBucket({
```

Comment:
I have no idea, but I had a wild guess that not awaiting this promise meant some files were not getting uploaded before discovery was starting, which could explain why only 2 files were found instead of 3. That really shouldn't be the case, but this isn't a harmful change either way

Reply:
Hmm, I don't think this does that - .map takes each element and returns an invocation of the internal method against each element in s3data, so in this case, that's still going to return an array of Promise objects that eventually resolve.... which is what the test was originally doing.

e.g.

```
> appendList = ['this', 'that']
[ 'this', 'that' ]
> someFunction  = async (bar) => `${bar}-append`
[AsyncFunction: someFunction]

> appendList.map((item) => someFunction(item))
[ Promise { 'this-append' }, Promise { 'that-append' } ]


> appendList.map(async (item) => await someFunction(item))
[ Promise { <pending> }, Promise { <pending> } ]


> await appendList.map(async (item) => await someFunction(item))
[ Promise { 'this-append' }, Promise { 'that-append' } ]


> await Promise.all(appendList.map((item) => someFunction(item)))
[ 'this-append', 'that-append' ]
>
```

I think this should be written like so: 

```
  await Promise.all(s3Data.map(async (file) => await updateAndUploadTestFileToBucket({
```

Reply:
Oh my. I can't believe I didn't see that. This seems very likely to be a cause of some integration test failures. 

🙈

Reply:
It's a really good catch, nevertheless. I'm sure I've glossed over that line at least a few times. 

I did a quick search for /await .*\.map/ and it doesn't appear that's elsewhere in this repo, so at least it's not a pattern that got by recent review.